IEX.h

```diff
@@ -85,310 +86,353 @@ namespace IEX {
-                ksd.day50MovingAvg      = getDouble(jsonData["day50MovingAvg"]);
-                ksd.institutionPercent  = getDouble(jsonData["institutionPercent"]);
-                ksd.insiderPercent      = getDouble(jsonData["insiderPercent"]);
-                ksd.shortRatio          = getDouble(jsonData["shortRatio"]);
-                ksd.year5ChangePercent  = getDouble(jsonData["year5ChangePercent"]);
-                ksd.year2ChangePercent  = getDouble(jsonData["year2ChangePercent"]);
-                ksd.year1ChangePercent  = getDouble(jsonData["year1ChangePercent"]);
-                ksd.ytdChangePercent    = getDouble(jsonData["ytdChangePercent"]);
-                ksd.month6ChangePercent = getDouble(jsonData["month6ChangePercent"]);
-                ksd.month3ChangePercent = getDouble(jsonData["month3ChangePercent"]);
-                ksd.month1ChangePercent = getDouble(jsonData["month1ChangePercent"]);
-                ksd.day5ChangePercent   = getDouble(jsonData["day5ChangePercent"]);
-                // clang-format on
+            PriceData pd;
+            pd.called_endpoint = url;
+            pd.stock_symbol = stock_symbol;
+            pd.latest_price = jsonData.asDouble();
 
-                return ksd;
-            }
-        };
-
-        struct CompanyData {
-            std::string called_endpoint;
-            std::string stock_symbol;
-            std::string company_name;
-            std::string exchange;
-            std::string industry;
-            std::string website;
-            std::string description;
-            std::string CEO;
-            std::string issue_type;
-            std::string sector;
-            std::vector<std::string> tags;
-        };
-
-        class Company {
-        public:
-            static std::size_t callback(const char* in,
-                                        std::size_t size,
-                                        std::size_t num,
-                                        std::string* out) {
-                const std::size_t totalBytes(size * num);
-                out->append(in, totalBytes);
-                return totalBytes;
-            }
+            return pd;
+        }
 
-            static CompanyData get(const std::string stock_symbol) {
-                std::string url = IEX_API_V1_ENDPOINT;
-                url += "/stock/" + std::string(stock_symbol) + "/company";
+        // Input:
+        //  stock_symbol - the stock symbol you want to query
+        //  time_period - the time period you want to look in.  Allowed
+        //  values are: 5y, 2y, 1y, ytd, 6m, 3m, and 1m
+        //
+        // Output:
+        //  Will returns a DividendsData struct with the relevant info
+        //  on success or an empty struct with the success field set to
+        //  false
+        static DividendsData getDividends(const std::string stock_symbol,
+                                         const std::string time_period) {
```

Comment:
Since this time period is well defined, I think this should be an enumeration so as to avoid any confusion if the wrong string is passed.

Reply:
I would agree with you, but we can't have enum values that start with numbers so 2y or 3m wouldn't be a valid enum value.

Reply:
We could add a "parameter validator" for this to make sure only valid methods are passed or just adding an error handler catching the HTTP 400 (bad request) exceptions coming from the server side. Doesn't need to be implemented now, we can just create an issue as "things to improve in the future".

Reply:
I think the way I have it set up is more appropriate.  For me, having a string option is easier to use than using an enum.